cfep-21.md

```diff
@@ -0,0 +1,60 @@
+
+<table>
+<tr><td> Title </td><td> Export license packages for static and header-only libraries </td>
+<tr><td> Status </td><td> Proposed </td></tr>
+<tr><td> Author(s) </td><td> Daniel Ching &lt;carterbox@users.noreply.github.com&gt;</td></tr>
+<tr><td> Created </td><td> Jul 9, 2022</td></tr>
+<tr><td> Updated </td><td> Jul 25, 2022</td></tr>
+<tr><td> Discussion </td><td> NA </td></tr>
+<tr><td> Implementation </td><td> NA </td></tr>
+</table>
+
+## Abstract
+
+This CFEP proposes that header-only and static libraries should be required
```

Comment:
Not all licenses require the distribution of the license when the code is distributed in binary form.

Reply:
Thanks for pointing this out. Common licenses like BSD3 and MIT do require distribution of the notice and or license with the binary form though. My thought is that it is easier to require all packages to attribute, than to have reviewers check against a running list of which licenses require attribution.

Reply:
Mmm, I think we should encourage people to be looking at and thinking about the licenses, rather than not. It shouldn't be hard to keep a list of licenses that do/don't have requirements, and if there are gray areas, I think we should be looking into them.

Reply:
I think we state that packages whose licenses require license file distribution must use a run export or other mechanism to do this. 

Others are free to not. There is no reason for us to require more work from folks when the underlying package does not require it.

Reply:
>  I think we should encourage people to be looking at and thinking about the licenses

This is a very good point, maintainers should understand the license of their packages, but then why keep a list of licenses that require attribution, that would discourage people from looking at the licenses because they would just look at the list.

> There is no reason for us to require more work from folks when the underlying package does not require it.

As I stated, I think it is more (human) work overall to determine on a case-by-case basis whether a license requires attribution than to just export a license package for every header library. I also don't see any harm in attributing when it is not required.

As a compromise, I propose that header-only libraries without license packages are required to have a comment that the license does not require attribution. Otherwise, it is unclear whether the maintainers have considered this CFEP.